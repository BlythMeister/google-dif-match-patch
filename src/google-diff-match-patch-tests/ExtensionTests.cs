using ApprovalTests;
using ApprovalTests.Reporters;
using DiffMatchPatch;
using NUnit.Framework;

namespace DiffMatchPatchTests
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class ExtensionTests
    {
        private const string expected = "start\n" +
                                        "line with some content\n" +
                                        "unchanged content\n" +
                                        "removed line\n" +
                                        "end";

        private const string actual = "start\n" +
                                      "new line\n" +
                                      "line with some modified content\n" +
                                      "unchanged content\n" +
                                      "end";

        [Test]
        public void CorrectPrettyTextOutput()
        {
            var dmp = new diff_match_patch();

            var diffs = dmp.diff_main(expected, actual);

            dmp.diff_cleanupForPrettyOutput(diffs);

            var (patchCount, prettyOutput) = dmp.diff_toPrettyText(diffs);

            Approvals.Verify(prettyOutput);
            Assert.AreEqual(3, patchCount);
        }

        [Test]
        public void CorrectPrettyHtmlOutput()
        {
            var dmp = new diff_match_patch();

            var diffs = dmp.diff_main(expected, actual);

            dmp.diff_cleanupForPrettyOutput(diffs);

            var prettyOutput = dmp.diff_toPrettyHtml(diffs);

            Approvals.Verify(prettyOutput);
        }
    }
}
