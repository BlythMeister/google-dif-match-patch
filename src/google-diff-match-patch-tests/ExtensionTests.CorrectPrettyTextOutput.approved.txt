
>> 1 ___________________________________
@@ -1,0 +1,0 @@
>> ---
[...]start
line wit[...]
~~~~~~~~~~~~~~~~~~~~
>> +++
[...]start
new line
line wit[...]
___________________________________ 1 <<

>> 2 ___________________________________
@@ -23,22 +23,22 @@
>> ---
[...]th some content[...]
~~~~~~~~~~~~~~~~~~~~
>> +++
[...]th some modified content[...]
___________________________________ 2 <<

>> 3 ___________________________________
@@ -62,61 +62,61 @@
>> ---
[...]ent
removed line
end[...]
~~~~~~~~~~~~~~~~~~~~
>> +++
[...]ent
end[...]
___________________________________ 3 <<

